concourse/pipeline.yml

```diff
@@ -35,65 +35,61 @@ jobs:
-                cf set-env govuk-coronavirus-business-volunteer-form BASIC_AUTH_PASSWORD "$BASIC_AUTH_PASSWORD"
-                cf set-env govuk-coronavirus-business-volunteer-form SENTRY_DSN "$SENTRY_DSN" 
-                cf set-env govuk-coronavirus-business-volunteer-form AWS_ACCESS_KEY_ID "$AWS_ACCESS_KEY_ID"
-                cf set-env govuk-coronavirus-business-volunteer-form AWS_SECRET_ACCESS_KEY "$AWS_SECRET_ACCESS_KEY"
-                cf set-env govuk-coronavirus-business-volunteer-form AWS_DYNAMODB_SUBMISSIONS_TABLE_NAME "$AWS_DYNAMODB_SUBMISSIONS_TABLE_NAME"
-                cf set-env govuk-coronavirus-business-volunteer-form SECRET_KEY_BASE "$SECRET_KEY_BASE"
-                cf v3-zdt-push govuk-coronavirus-business-volunteer-form --wait-for-deploy-complete --no-route
-                cf map-route govuk-coronavirus-business-volunteer-form cloudapps.digital --hostname govuk-coronavirus-business-volunteer-form-stg
+        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/deploy-to-govuk-paas.yml
+        params:
+          CF_SPACE: staging
+          INSTANCES: 5
+          CF_STARTUP_TIMEOUT: 5 # minutes
+          REQUIRE_BASIC_AUTH: "true"
+          BASIC_AUTH_PASSWORD: ((basic-auth-password))
+          AWS_ACCESS_KEY_ID: ((aws-access-key-id))
+          AWS_SECRET_ACCESS_KEY: ((aws-secret-access-key))
+          SECRET_KEY_BASE: ((secret-key-base-staging))
+          AWS_DYNAMODB_SUBMISSIONS_TABLE_NAME: coronavirus-business-staging
+          HOSTNAME: govuk-coronavirus-business-volunteer-form-stg
+          SENTRY_DSN: ((sentry-dsn))
 
   - name: smoke-test-staging
     plan:
       - get: govuk-coronavirus-business-volunteer-form
         trigger: true
         passed: [deploy-to-staging]
       - task: smoke-test
-        config:
-          platform: linux
-          image_resource:
-            type: docker-image
-            source:
-              repository: governmentpaas/curl-ssl
-          run:
-            # TODO we can come up with a more thorough test than this
-            path: curl
-            args: ['--fail', '--silent', '--verbose', '--user', 'gds:((basic-auth-password))', 'https://govuk-coronavirus-business-volunteer-form-stg.cloudapps.digital/']
+        file: govuk-coronavirus-vulnerable-people-form/concourse/tasks/smoke-test.yml
+        params:
+          URL: 'https://govuk-coronavirus-business-volunteer-form-stg.cloudapps.digital/'
+          MESSAGE: "Checks that the application deployed to staging is not serving HTTP error codes"
 
+  - name: deploy-to-prod
+    serial: true
+    plan:
+      - get: govuk-coronavirus-business-volunteer-form
+        trigger: true
+        passed: [smoke-test-staging]
+      - task: deploy-to-paas
+        file: govuk-coronavirus-business-volunteer-form/concourse/tasks/deploy-to-govuk-paas.yml
+        params:
+          CF_SPACE: production
+          INSTANCES: 30
+          CF_STARTUP_TIMEOUT: 15 # minutes
+          REQUIRE_BASIC_AUTH:
```

Comment:
It looks like `REQUIRE_BASIC_AUTH: "true"` isn't true for prod. Do we want that?

Reply:
It will default to falsy

Reply:
When we went live with the other form we had basic auth on for about 5 mins after we mapped the route. Probably not worth it.

Reply:
It might be nice to have basic auth enabled if we want to test the DB etc is working in prod before we release.